### PR TITLE
feat: add ethereum methods in EthereumProvider

### PR DESCRIPTION
**Pull request by lucasgabrielgsp**
- [x] Add ethereum methods in  EthereumProvider file.

### CODE REVIEW

File: source/scripts/Provider/EthereumProvider.ts [hunk 1]
```diff
@@ -1,28 +1,49 @@
...
-    getChainId,
-    getAddress,
+    createAccount,
+    importAccount,
+    sendTransactions,
+    changeNetwork,
+    getNFTImage,
+    getUserNFT,
+    getTokenIconBySymbol,
+    web3Provider,
```
Comment (amandagonsalves) [inline source/scripts/Provider/EthereumProvider.ts:45]: we can remove this as well

File: package.json [hunk 2]
```diff
@@ -47,6 +47,10 @@
     "@headlessui/react": "^1.4.2",
     "@heroicons/react": "^1.0.5",
     "@reduxjs/toolkit": "^1.4.0",
+    "@syspollum/sysweb3-core": "^1.0.2",
```
Comment (amandagonsalves) [inline package.json:50]: we can now change the scope to @pollum-io

File: source/scripts/Provider/EthereumProvider.ts [hunk 3]
```diff
@@ -1,28 +1,49 @@
...
+import { web3Provider as provider } from '@syspollum/sysweb3-network';
 
-  const getBalance = () => {};
-
-  const getXpub = () => {};
-
-  const getTokens = () => {};
-
-  const getState = () => {};
+export const EthereumProvider = () => {
```
Comment (amandagonsalves) [inline source/scripts/Provider/EthereumProvider.ts:8]: we need some more methods available to websites like

- get address
- get network
- get state
- get accounts
- get chain id for the current provider

File: source/scripts/Provider/EthereumProvider.ts [hunk 4]
```diff
@@ -1,28 +1,49 @@
...
+
+  const getBalance = (walletAddress: string) =>
+    Web3Accounts().getBalance(walletAddress);
+
+  const changeNetwork = (networkId: number) =>
+    Web3Accounts().setActiveNetwork(networkId);
+
+  const getNFTImage = (NFTContractAddress: string, tokenId: number) =>
+    web3NFTImage(NFTContractAddress, tokenId);
+  const getUserNFT = (walletAddress: string) =>
```
Comment (amandagonsalves) [inline source/scripts/Provider/EthereumProvider.ts:30]: we can remove this and use the method available in @pollum-io/sysweb3-utils, as well as getTokenIconBySymbol and any other utility function that we can expose to the website directly from our lib

File: source/scripts/Provider/EthereumProvider.ts [hunk 5]
```diff
@@ -1,28 +1,37 @@
...
+  };
 
-  const getTokens = () => {};
+  const getAddress = async () => {
+    const address = await web3Provider.eth.getAccounts();
+    return address[0];
+  };
 
-  const getState = () => {};
+  const getBalance = async (address: string) => {
```
Comment (lucas-gauer) [inline source/scripts/Provider/EthereumProvider.ts:25]: If all we do is point to the function, can be written like:
```
  const getBalance = web3Provider.eth.getBalance;
```

**Review by lucas-gauer — CHANGES_REQUESTED** [review]

**Comment by lucas-gauer** [comment]
What about the changes that i requested?

**Pull request merged**